src/backend-c/runtime/cast_queue.c

```diff
@@ -0,0 +1,69 @@
+#include "assert.h"
+#include "cast_queue.h"
+
+
+cast_queue* allocate_cast_queue() {
+  cast_queue *cq = GC_NEW(cast_queue);
+  cq->casts = GC_MALLOC(sizeof(suspended_cast[CAST_QUEUE_INIT_CAPACITY]));
+  cq->capacity = CAST_QUEUE_INIT_CAPACITY;
+  cq->front = 0;
+  cq->rear = 0;
+  cq->size = 0;
```

Comment:
I added an extra field to track the size. You could compute the size on the fly with a few checks but I opted to use the extra field instead because enqueue needs that information in each call.